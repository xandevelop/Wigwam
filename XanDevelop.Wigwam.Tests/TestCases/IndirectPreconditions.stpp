test | i can log in
open login page
submit form

test | i can email admin
open contact page
submit form

func | open login page
echo | login page opened
post | page | login

func | open contact page
echo | contact page opened
post | page | contact

func | submit form
echo | submitting form
click submit

func | click submit
pre | page | login
echo | clicked submit on login page

func | click submit
pre | page | contact
echo | clicked submit on contact page