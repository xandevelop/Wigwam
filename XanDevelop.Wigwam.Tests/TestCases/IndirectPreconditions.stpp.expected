Start Test i can log in
    Start Func Call open login page
    End Func Call open login page
    Start Func Call submit form
    End Func Call submit form
End Test i can log in

Start Test i can email admin
    Start Func Call open contact page
    End Func Call open contact page
    Start Func Call submit form
    End Func Call submit form
End Test i can email admin

Start Function open login page
    Command | command=echo | target=login page opened | value=
End Function open login page

Start Function open contact page
    Command | command=echo | target=contact page opened | value=
End Function open contact page

Start Function submit form
    Command | command=echo | target=submitting form | value=
    Start Func Call click submit
    End Func Call click submit
End Function submit form

Start Function click submit
PRE: page Equals login
    Command | command=echo | target=clicked submit on login page | value=
End Function click submit

Start Function click submit
PRE: page Equals contact
    Command | command=echo | target=clicked submit on contact page | value=
End Function click submit

Start Function submit form
PRE: page Equals contact
    Command | command=echo | target=submitting form | value=
    Start Func Call click submit
    End Func Call click submit
End Function submit form